src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,46 @@
 
 class Animal {
   // write your code here
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+    this.addBeast(this);
+  }
+  hide() {
+    this.hidden = true;
+    this.addBeast(this);
```

Comment:
it is definitely redundant here
no need to add a new animal every time when it hides